src/openslide-vendor-generic-tiff.c

```diff
@@ -175,6 +175,20 @@ static int width_compare(gconstpointer a, gconstpointer b) {
   }
 }
 
+static void set_resolution_prop(openslide_t *osr, TIFF *tiff,
+                                const char *property_name,
+                                ttag_t tag) {
+  float f;
+  uint16_t unit;
+
+  if (TIFFGetFieldDefaulted(tiff, TIFFTAG_RESOLUTIONUNIT, &unit) &&
+      TIFFGetField(tiff, tag, &f) &&
+      unit == RESUNIT_CENTIMETER) {
```

Comment:
Since this is a generic driver, we should probably also handle the conversion if the unit is inches.